src/components/ProductVersionBuildConfigsEditPage/ConfigsAddList.tsx

```diff
@@ -80,6 +86,12 @@ export const ConfigsAddList = <T extends BuildConfiguration | GroupConfiguration
     (config: T) =>
       addedConfigs.every((addedConfig) => addedConfig.id !== config.id) &&
       (!productVersionToExclude || config.productVersion?.id !== productVersionToExclude) &&
+      (!buildConfigToExclude || !isBuildConfig(config) || config.id !== buildConfigToExclude) &&
```

Comment:
Please create some named variable explaining individual condition groups, or at least some TSDocs would be helpful. Now it's very hard to maintain.